### ISSUE TITLE
NPE when running integration

### ISSUE THREAD

**Issue by rhuss** [atlasmap/camel-atlasmap#14]
When running a simple integration 

```yml
---
flows:
- steps:
  - kind: endpoint
    uri: twitter-mention
  - kind: endpoint
    uri: atlas:mapping-step-2.json
  - kind: endpoint
    uri: salesforce-upsert-contact
```

with a mapping

```json
{
  "AtlasMapping": {
    "jsonType": "io.atlasmap.v2.AtlasMapping",
    "dataSource": [
      {
        "jsonType": "io.atlasmap.v2.DataSource",
        "id": "twitter4j.Status",
        "uri": "atlas:java?className=twitter4j.Status",
        "dataSourceType": "SOURCE"
      },
      {
        "jsonType": "io.atlasmap.v2.DataSource",
        "id": "io.syndesis.connector.salesforce.Contact",
        "uri": "atlas:java?className=io.syndesis.connector.salesforce.Contact",
        "dataSourceType": "TARGET"
      }
    ],
    "mappings": {
      "mapping": [
        {
          "jsonType": "io.atlasmap.v2.Mapping",
          "mappingType": "MAP",
          "inputField": [
            {
              "jsonType": "io.atlasmap.java.v2.JavaField",
              "name": "screenName",
              "path": "/user/screenName",
              "fieldType": "STRING",
              "docId": "twitter4j.Status"
            }
          ],
          "outputField": [
            {
              "jsonType": "io.atlasmap.java.v2.JavaField",
              "name": "TwitterScreenName__c",
              "path": "/TwitterScreenName__c",
              "fieldType": "STRING",
              "docId": "io.syndesis.connector.salesforce.Contact"
            }
          ]
        }
      ]
    },
    "name": "UI.353284",
    "lookupTables": {
      "lookupTable": []
    },
    "properties": {
      "property": []
    }
  }
}

```

then this NPE happens

```
io.atlasmap.api.AtlasException: java.lang.NullPointerException
	at io.atlasmap.java.module.JavaModule.processInputMapping(JavaModule.java:190)
	at io.atlasmap.core.DefaultAtlasContext.process(DefaultAtlasContext.java:180)
	at org.apache.camel.component.atlasmap.AtlasEndpoint.onExchange(AtlasEndpoint.java:193)
	at org.apache.camel.impl.ProcessorEndpoint$1.process(ProcessorEndpoint.java:71)
	at org.apache.camel.util.AsyncProcessorConverterHelper$ProcessorToAsyncProcessorBridge.process(AsyncProcessorConverterHelper.java:61)
	at org.apache.camel.processor.SendProcessor.process(SendProcessor.java:145)
	at org.apache.camel.management.InstrumentationProcessor.process(InstrumentationProcessor.java:77)
	at org.apache.camel.processor.RedeliveryErrorHandler.process(RedeliveryErrorHandler.java:541)
	at org.apache.camel.processor.CamelInternalProcessor.process(CamelInternalProcessor.java:198)
	at org.apache.camel.processor.Pipeline.process(Pipeline.java:120)
	at org.apache.camel.processor.Pipeline.process(Pipeline.java:83)
	at org.apache.camel.processor.CamelInternalProcessor.process(CamelInternalProcessor.java:198)
	at org.apache.camel.processor.DelegateAsyncProcessor.process(DelegateAsyncProcessor.java:110)
	at org.apache.camel.component.connector.ConnectorConsumerProcessor.process(ConnectorConsumerProcessor.java:66)
	at org.apache.camel.processor.DelegateAsyncProcessor.process(DelegateAsyncProcessor.java:97)
	at org.apache.camel.component.twitter.consumer.TwitterConsumerPolling.poll(TwitterConsumerPolling.java:64)
	at org.apache.camel.impl.ScheduledPollConsumer.doRun(ScheduledPollConsumer.java:174)
	at org.apache.camel.impl.ScheduledPollConsumer.run(ScheduledPollConsumer.java:101)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException: null
	at io.atlasmap.java.module.JavaModule.resolveGetMethod(JavaModule.java:410)
	at io.atlasmap.java.module.JavaModule.populateSourceFieldValue(JavaModule.java:220)
	at io.atlasmap.java.module.JavaModule.processInputMapping(JavaModule.java:211)
	at io.atlasmap.java.module.JavaModule.processInputMapping(JavaModule.java:182)
	... 24 common frames omitted
```

**Comment by rhuss** [comment]
Not sure whether this is an atlasmap or a camel-atlasmap issue. 

@igarashitm could you have look please ?

**Comment by igarashitm** [comment]
At least NPE with a message 'null' doesn't look right :) will do after lunch.

**Comment by igarashitm** [comment]
Here is a unit test without camel
https://github.com/igarashitm/atlasmap/commit/884fd510166582dd49d0427d1eeff38072d1131a

So what happens is that if ..mapping/inputField/@docId is specified, JavaModule can't hit the input object in the map here:
https://github.com/atlasmap/atlasmap/blob/master/atlas-java-parent/atlas-java-module/src/main/java/io/atlasmap/java/module/JavaModule.java#L176-L179
As the input is only mapped to the default key, AtlasConstants.DEFAULT_SOURCE_DOC_ID.

Now the question is how should it be,
1. If docId is specified, the input should be mapped to the docId in the input map
2. The docId should not be specified or require additional entry in the mapping

WDYT @mattrpav ?

@rhuss so quick hack is to just remove the docId from mapping def, but I guess you're trying it via UI so we need to know how it really should be...

**Comment by igarashitm** [comment]
Fixed - atlasmap/atlasmap#159
